src/lightning_lite/wrappers.py

```diff
@@ -160,6 +161,13 @@ def __len__(self) -> int:
         return len(self._dataloader)
 
     def __iter__(self) -> Union[Iterator[Any], Generator[Any, None, None]]:
+        if hasattr(self._dataloader.sampler, "set_epoch"):
+            # Without setting the epoch, the distributed sampler would return the same indices every time, even when
+            # shuffling is enabled. In PyTorch, the user would normally have to call `.set_epoch()` on the sampler.
+            # In Lite, we take care of this boilerplate code.
+            self._dataloader.sampler.set_epoch(self._epoch)
```

Comment:
Neat !

Reply:
Should we add a log info there ? So users don't set_epoch too ?

Reply:
if we do, it should at max be a log debug :) Otherwise this will become very verbose

Reply:
I don't think there is a need to inform. We also don't do it in Lightning Trainer.
It's fine if the user also calls `set_epoch` manually.